hobo/src/world.rs

```diff
@@ -74,6 +74,26 @@ unsafe impl Send for World {}
 unsafe impl Sync for World {}
 
 impl World {
+	#[cfg(debug_assertions)]
+	#[track_caller]
+	pub(crate) fn dyn_storage<Component: 'static>(&self) -> std::cell::Ref<'static, Box<dyn DynStorage>> {
+		let caller = std::panic::Location::caller();
+
+		if let Some(storage) = self.storages.map_get(&TypeId::of::<Component>(), |x| x.try_borrow()) {
+			storage.unwrap_or_else(|e| panic!("'{e}': Immutably borrowed `{}` storage at `{caller}` while a mutable borrow to it already exists:\n\n{}\n",
```

Comment:
I'd rephrase the error message slightly: `Trying to immutably borrow ... while a mutable borrow ...`